Jobsway2goMvc/Controllers/JobCategoriesController.cs

```diff
@@ -151,20 +151,26 @@ public async Task<IActionResult> Delete(int? id)
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> DeleteConfirmed(int id)
         {
-            if (_context.JobCategories == null)
+            if(_context.Jobs == null)
             {
-                return Problem("Entity set 'ApplicationDbContext.JobCategories'  is null.");
+                return Problem("Entity set 'ApplicationDbContext.Jobs'  is null.");
             }
-            var jobCategory = await _context.JobCategories.FindAsync(id);
-            if (jobCategory != null)
+
+            var jobs = _context.Jobs.Where(x => x.CategoryId == id);
+            //var category = _context.JobCategories.Where(x => x.Id == id);
+ 
+            if (jobs.Count() > 0)
             {
-                _context.JobCategories.Remove(jobCategory);
+                ViewBag.Delete = "Cannot delete Job Category because it has jobs assigned to it.";
+                return View();
+            }
+            else
+            {
+                _context.JobCategories.Remove(category.SingleOrDefault());
```

Comment:
`category.SingleOrDefault()` doesn't exist anymore as it is deleted from above.
Find category with Single if id is not null or 0 and then use that reference to remove it.